### ISSUE TITLE
`s3v2` seems to ignore input cfgs, using just defaults

### ISSUE THREAD

**Issue by shlomi-dr** [xitongsys/parquet-go-source#43]
Trying to use the new `s3v2` apis, I started getting error messages that "I did not specify regions", though I did. Debugging lead me to the following, where the `cfgs` defined on line 71  are silently ignored by the body:

https://github.com/xitongsys/parquet-go-source/blob/b4f792c43cd0b2477327d56a74970efca07af04e/s3v2/s3.go#L66-L80

Am I doing something wrong?

**Comment by shlomi-dr** [comment]
Worked around this by using `NewS3FileWriterWithClient` directly. 

Really confused why `NewS3FileWriter` has such misleading signature..

**Comment by sjtower** [comment]
The same problem exists for `func NewS3FileReader`:
`// NewS3FileReader creates an S3 FileReader, to be used with NewParquetReader
func NewS3FileReader(ctx context.Context, bucket string, key string, cfgs ...*aws.Config) (source.ParquetFile, error) {
	return NewS3FileReaderWithClient(ctx, s3.NewFromConfig(getConfig()), bucket, key)
}`

the input parameter `cfgs` is not used. Shouldn't there only be one configuration passed?

EG:

`// NewS3FileReader creates an S3 FileReader, to be used with NewParquetReader
func NewS3FileReader(ctx context.Context, bucket string, key string, cfg aws.Config) (source.ParquetFile, error) {
	return NewS3FileReaderWithClient(ctx, s3.NewFromConfig(cfg), bucket, key)
}`

**Comment by shlomi-dr** [comment]
Yeah, don't know why these functions take `n` cfgs, and use `0` 🤷‍♂️  

For now I just work with the `*WithClient` functions.
